File: app/src/main/java/info/jukov/vkstoryteller/surface/DragableImage.java
package info.jukov.vkstoryteller.surface;

import android.graphics.Bitmap;
import android.graphics.Rect;

/**
 * User: jukov
 * Date: 06.09.2017
 * Time: 21:11
 */

public final class DragableImage {

	private final Bitmap bitmap;
	private final int width;
	private final int height;
	private float x;
	private float y;
	private float scale = 1;
	private float angle = 0;

	public DragableImage(final Bitmap bitmap) {
		this.bitmap = bitmap;
		this.width = bitmap.getWidth();
		this.height = bitmap.getHeight();
	}

	public Bitmap getBitmap() {
		return bitmap;
	}

	public int getWidth() {
		return width;
	}

	public int getHeight() {
		return height;
	}

	public float getX() {
		return x;
	}

	public float getY() {
		return y;
	}

	public void setX(final float x) {
		this.x = x;
	}

	public void setY(final float y) {
		this.y = y;
	}

	public float getScale() {
		return scale;
	}

	public void setScale(final float scale) {
		this.scale = scale;
	}

	public float getAngle() {
		return angle;
	}

	public void setAngle(final float angle) {
		this.angle = angle;
	}

	public int getWidthCenter() {
		return width / 2;
	}

	public int getHeightCenter() {
		return height / 2;
	}

	public Rect getRect() {
		return new Rect((int) (x + scale), (int) (y + scale), (int) (x + width - scale), (int) (y + height - scale));
	}
}

File: app/src/main/java/info/jukov/vkstoryteller/surface/DragableImage.kt
package info.jukov.vkstoryteller.surface

import android.graphics.Bitmap

/**
 * User: jukov
 * Date: 10.09.2017
 * Time: 13:26
 */
data class DragableImage(
        val bitmap: Bitmap,
        var x: Float = 0.0f,
        var y: Float = 0.0f,
        var scale: Float = 1.0f,
        var angle: Float = 0.0f) {

    val width = bitmap.width;
    val height = bitmap.height;
    val widthCenter = bitmap.width / 2.0f;
    val heightCenter = bitmap.height / 2.0f;

}
